### PR TITLE
Second round with updated syntax and file distribution

### PR DESCRIPTION
**Pull request by FedG-code**

### CODE REVIEW

File: blogs/converter.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+const {readFileSync} = require('fs');
+const {writeFileSync} = require('fs');
+
+const {marked} = require('marked');
+
+const listOfMarkdownFiles = readdirSync('markdown');
+
+console.log(listOfMarkdownFiles);
+
+listOfMarkdownFiles.forEach(function(element){
```
Comment (robinglen) [inline blogs/converter.js]: you can do this in the new way with arrow functions

```
listOfMarkdownFiles.forEach(element => {
...
})
```
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Arrow_functions

File: blogs/converter.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
+const {readdirSync} = require('fs');
```
Comment (robinglen) [inline blogs/converter.js]: you can do this as one line, for each object
```
const {readdirSync, readFileSync, writeFileSync} = require('fs');
```

File: blogs/converter.js [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+
+const listOfMarkdownFiles = readdirSync('markdown');
+
+console.log(listOfMarkdownFiles);
+
+listOfMarkdownFiles.forEach(function(element){
+    const fileContent = readFileSync(`markdown/${element}`, 'utf8');
+    //Turn content to string
+   //pass the file through marked
+   var newHTMLFile = marked(fileContent);
```
Comment (robinglen) [inline blogs/converter.js]: this can be a `const`

File: blogs/converter.js (hunk 1 above)
Comment (robinglen) [reply blogs/converter.js]: I would also consider something more descriptive for the variable name like `file`

File: blogs/converter.js [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+   //get the new file name
+   const newFileName = element.replace('.md', '.html');
+   //write the html file with its new name
+   writeFileSync(newFileName, newHTMLFile);
```
Comment (robinglen) [inline blogs/converter.js]: This is stylist or you can say being "clean" but it doesn't really matter but as you're not using these variables more than once, you can do this

```
writeFileSync(element.replace('.md', '.html'), marked(fileContent));
```

Some people don't like that as what you have is easier to read... so its not critical but if youre not using a variable more than once for memory reasons should you even set it? To be honest the garbage collector solves all of this... but just something good for you to know

**Review by robinglen — COMMENTED** [review]
some more things to make it even cleaner

File: blogs/converter.js (hunk 1 above)
Comment (FedG-code) [reply blogs/converter.js]: So that's a variable? You can make variables without even using var or const?

Comment (robinglen) [reply blogs/converter.js]: so youre not declaring a variable, its a value being return by the method `forEach`

File: blogs/converter.js [hunk 5]
```diff
@@ -0,0 +1,20 @@
+const {readdirSync, readFileSync, writeFileSync} = require('fs');
+
+var showdown = require('showdown');
```
Comment (robinglen) [inline blogs/converter.js]: you can `const` this

File: blogs/converter.js [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+
+var convertToHTML = new showdown.Converter();
```
Comment (robinglen) [inline blogs/converter.js]: `const` here too

File: blogs/package.json [hunk 7]
```diff
@@ -0,0 +1,17 @@
...
+  "main": "converter.js",
+  "scripts": {
+    "test": "echo \"Error: no test specified\" && exit 1",
+    "start": "node converter.js"
+  },
+  "keywords": [],
+  "author": "",
+  "license": "ISC",
+  "dependencies": {
+    "marked": "^4.1.0",
```
Comment (robinglen) [inline blogs/package.json]: Do you want to use marked or showed? Remove the one you don't want to use from here, then do another 
```
npm i
```

File: blogs/markdown/microservices_solving-a-problem-like-routine_2020-update.md [hunk 8]
```diff
@@ -1,4 +1,7 @@
-﻿# Microservices: Solving a problem like routing — 2020 update
+﻿
```
Comment (robinglen) [inline blogs/markdown/microservices_solving-a-problem-like-routine_2020-update.md]: we removing these lines?

File: blogs/converter.js [hunk 9]
```diff
@@ -0,0 +1,20 @@
...
+
+listOfMarkdownFiles.forEach(file=>{
+    const fileContent = readFileSync(`markdown/${file}`, 'utf8');
+
+    const titleFormatting = '\n\n\n';
+    const formattedFile = (titleFormatting + fileContent);
+   //pass the file through marked
+   const newHTMLFile = convertToHTML.makeHtml(formattedFile);
+   //write the html file with its new name
+   writeFileSync(file.replace('.md', '.html'), newHTMLFile);
```
Comment (robinglen) [inline blogs/converter.js]: will this put it in the `blogs` directory? We should also rename that directory to 'blog'

Comment (FedG-code) [reply blogs/converter.js]: yeah because the code is in the blog directory

**Review by robinglen — APPROVED** [review]

**Pull request merged**